### PR TITLE
Update filter code to include upstream changes, fix filter property type

### PR DESCRIPTION
**Pull request by avaleske**
## Issue #143 

Merge in upstream changes to filters.

Addresses #143 
- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description
Merges in changes to upstream filters. See https://github.com/bloom-housing/bloom/pull/1359/files for changed filed.
- Adds generic filters query param to listings.controller.ts
- Adds generic filters handling to listings.service.ts
- Adds an enum of filter types to listing.dto.ts, so we can handle specific filters differently
- Changes upstream filter param from `ListingFilterParam[]` to `ListingFilterParam` to match its actual type.

I opened issues to for some TODOs in this PR:
- #202 
- #210 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [x] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

Test with paths like
- `/listings?filter[$comparison]=<>&filter[neighborhood]=Foster%20City`
- `/listings?filter[$comparison]=<>&filter[name]=colesium&filter[$comparison]==&filter[neighborhood]=Foster%20City`

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [x] I have updated the changelog to include a description of my changes

### CODE REVIEW

File: backend/core/src/listings/listings.controller.ts [hunk 1]
```diff
@@ -36,17 +37,19 @@ import { ValidationsGroupsEnum } from "../shared/types/validations-groups-enum"
...
-    type: String,
-    example: "Fox Creek",
+    name: "filter",
     required: false,
-    description: "The neighborhood to filter by",
+    type: [String],
+    items: {
+      $ref: getSchemaPath(ListingFilterParams),
+    },
+    example: { $comparison: ["=", "<>"], status: "active", name: "Coliseum" },
```
Comment (willrlin) [inline backend/core/src/listings/listings.controller.ts:49]: I take it at least for now we're going to continue using the `filter[$comparison]=<>&filter[name]=Coliseum` style? Can we start a conversation with Exygy about this? It doesn't feel very user-friendly/intuitive.

File: backend/core/src/listings/listings.service.ts [hunk 2]
```diff
@@ -35,21 +37,16 @@ export class ListingsService {
       .leftJoinAndSelect("listings.reservedCommunityType", "reservedCommunityType")
   }
 
-  public async list(
-    origin: string,
-    params: ListingsListQueryParams
-  ): Promise<PaginatedListingsDto> {
-    let query = this.getQueryBuilder().orderBy({
+  public async list(origin: string, params: ListingsQueryParams): Promise<PaginatedListingsDto> {
+    let qb = this.getQueryBuilder()
```
Comment (willrlin) [inline backend/core/src/listings/listings.service.ts:41]: Maybe this is a nit, but upstream has `let query = this.getQueryBuilder()`.  If you could submit a rename commit upstream, then we can keep these unified and not have worse merge conflicts.

Comment (avaleske) [reply backend/core/src/listings/listings.service.ts:41]: Are you sure? It looks like upstream is `qb`:
https://github.com/bloom-housing/bloom/blob/0b1cace2d75ef3d64a01f67a22185366ee9bc48d/backend/core/src/listings/listings.service.ts#L36

File: backend/core/src/listings/listings.controller.ts (hunk 1 above)
Comment (avaleske) [reply backend/core/src/listings/listings.controller.ts:49]: Once I realized that it's just object syntax it made sense, but I agree that it's confusing (and that URLs are a part of UX). Fwiw, it's a standard serialization format supported by Open API called Deep Object: https://swagger.io/docs/specification/serialization/

I asked [here](https://github.com/bloom-housing/bloom/discussions/1400#discussioncomment-1002553) about their future plans.

Comment (avaleske) [reply backend/core/src/listings/listings.controller.ts:49]: But for now, yes, I figured we'd leave it this way.

File: backend/core/src/listings/listings.service.ts (hunk 2 above)
Comment (willrlin) [reply backend/core/src/listings/listings.service.ts:41]: You're right! Maybe I was looking at the comparison in the wrong direction or something.

**Review by elenm — APPROVED** [review]

**Pull request merged**